### PR TITLE
Add option to extension settings to disable language server

### PR DESCRIPTION
**Pull request by ReeceHumphreys**

### CODE REVIEW

File: client/src/extension.ts [hunk 1]
```diff
@@ -35,24 +35,52 @@ const createClient = (
...
-      client.sendNotification("workspace/didChangeConfiguration", {
-        settings: {
-          slice: { sourceDirectory, referenceDirectories },
-        },
-      });
+      // Send the updated configuration to the language server
+      if (client) {
+        client.sendNotification("workspace/didChangeConfiguration", {
+          settings: {
+            slice: { enableLanguageServer, referenceDirectories },
```
Comment (externl) [inline client/src/extension.ts:64]: I don't think we need to send this setting to the language server. Does it care if it's disabled?

File: client/src/extension.ts [hunk 2]
```diff
@@ -35,24 +35,52 @@ const createClient = (
...
+          },
+        });
+      }
+
+      // Handle the enabling/disabling of the language server
+      if (event.affectsConfiguration("slice.languageServer.enabled")) {
+        if (enableLanguageServer && !client) {
+          traceOutputChannel.appendLine(
+            "Enabling language server as per configuration change..."
```
Comment (externl) [inline client/src/extension.ts:73]:
```suggestion
            "Enabling language server..."
```

File: client/src/extension.ts [hunk 3]
```diff
@@ -35,24 +35,52 @@ const createClient = (
...
+          );
+          await restartLanguageServer(context);
+        } else if (!enableLanguageServer && client) {
+          traceOutputChannel.appendLine(
+            "Disabling language server as per configuration change..."
```
Comment (externl) [inline client/src/extension.ts:78]:
```suggestion
            "Disabling language server..."
```

File: server/src/main.rs [hunk 4]
```diff
@@ -127,6 +127,14 @@ impl LanguageServer for Backend {
             .log_message(MessageType::INFO, "Slice Language Server config changed")
             .await;
 
+        // Check if disableLanguageServer is set to true
+        if !self.should_enable_language_server(params.clone()).await {
```
Comment (externl) [inline server/src/main.rs:131]: Can this case actually happen?

**Review by externl — APPROVED** [review]

File: server/src/main.rs (hunk 4 above)
Comment (ReeceHumphreys) [reply server/src/main.rs:131]: I think your right, not seeinng the logs on the server so its shutdown before this even fires.

**Pull request merged**